File: Livro-FundamentosDaProgramacaoDeComputadores/Capitulo3/C/Exercicio23.c
/*23. Faça um programa que receba um número real, encontre e mostre:
a) a parte inteira desse número;
b) a parte fracionária desse número;
c) o arredondamento desse número.*/

#include <stdio.h>
#include <math.h>

int main()
{
    float numero, inteira = 0, fracionaria = 0, arredondada = 0;

    printf("Digite o numero: ");
    scanf("%f", &numero);

    inteira = floor(numero);
    fracionaria = numero - inteira;
    arredondada = round(numero);

    printf("Parte inteira: %f\n", inteira);
    printf("Parte fracionaria: %f\n", fracionaria);
    printf("Parte arredondada: %f\n", arredondada);

    return 0;
}

File: Livro-FundamentosDaProgramacaoDeComputadores/Capitulo3/Java/Exercicio23.java
/*23. Faça um programa que receba um número real, encontre e mostre:
a) a parte inteira desse número;
b) a parte fracionária desse número;
c) o arredondamento desse número.*/

import java.util.Scanner;

public class Exercicio23 {
    public static void main(String[] args){
        Scanner scan = new Scanner(System.in);
        
        System.out.print("Digite um número real: ");
        double numero = scan.nextDouble();

        double inteira = Math.floor(numero);
        double fracionaria = numero - inteira;
        double arredondada = Math.round(numero);

        System.out.printf("\nParte inteira: %f", inteira);
        System.out.printf("\nParte fracionária: %f", fracionaria);
        System.out.printf("\nParte arredondada: %f\n", arredondada);

        scan.close();
    }
    
}
